### PR TITLE
feat:Objects and Array(#39517)

### PR DESCRIPTION
**Pull request by ajayrose25**

### CODE REVIEW

File: extra/utils/permissions.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const permissions = {
+    'getUsers': {
+    all: ['head-trainer'],
+    read : ['trainee', 'trainer'],
+    write : ['trainer'],
+    delete: [],
+    }
+    }
+    function haspermission(modulename,role,permissiontype)
```
Comment (VisaaliSuccessive) [inline extra/utils/permissions.js]: Please use camel casing for variable and function names

File: extra/utils/permissions.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        let roles=modulename[permissiontype];
```
Comment (VisaaliSuccessive) [inline extra/utils/permissions.js]: give spaces before and after equal to sign

File: extra/utils/validation.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+let c=0,d=0;
```
Comment (VisaaliSuccessive) [inline extra/utils/validation.js]: remove unused variables declared

File: extra/utils/validation.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    reviewerEmail : "reviewer2@successive.tech"},
+
+{
+    traineeEmail : "trainee3@successive.tech",
+    reviewerEmail : "reviewer3@successive.tech"},
+
+{
+traineeEmail : "trainee4@successive.tech",
+reviewerEmail : "reviewer4@successive.tech"},
+];
```
Comment (VisaaliSuccessive) [inline extra/utils/validation.js]: indent the code properly

File: extra/utils/validation.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+reviewerEmail : "reviewer4@successive.tech"},
+];   
+function ValidateEmail(mail) 
+{
+ let ab=/([a-zA-Z0-9])+@(successive)\.(tech)/g;
+   if(ab.test(mail)) return true 
+   else return false
+}
+
+function validateusers(users){
```
Comment (VisaaliSuccessive) [inline extra/utils/validation.js]: Use camel casing for variable and function names

File: extra/utils/permissions.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+const permissions = {
```
Comment (VisaaliSuccessive) [inline extra/utils/permissions.js]: Do the necessary commenting

File: extra/utils/validation.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+let c=0,d=0;
+const users=[ {
```
Comment (VisaaliSuccessive) [inline extra/utils/validation.js]: Do the necessary commenting

File: extra/utils/permissions.js (hunk 1 above)
Comment (ajayrose25) [reply extra/utils/permissions.js]: Done

File: extra/utils/permissions.js (hunk 2 above)
Comment (ajayrose25) [reply extra/utils/permissions.js]: Done

File: extra/utils/validation.js (hunk 3 above)
Comment (ajayrose25) [reply extra/utils/validation.js]: Done

File: extra/utils/validation.js (hunk 4 above)
Comment (ajayrose25) [reply extra/utils/validation.js]: Done

File: extra/utils/validation.js (hunk 5 above)
Comment (ajayrose25) [reply extra/utils/validation.js]: Done

File: extra/utils/permissions.js (hunk 6 above)
Comment (ajayrose25) [reply extra/utils/permissions.js]: Done

File: extra/utils/validation.js (hunk 7 above)
Comment (ajayrose25) [reply extra/utils/validation.js]: Done

File: extra/utils/permissions.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+const permissions = {     //Permissions is a object having various arrays and it is used to check all operations a person can perform
+    'getUsers': {
+    all: ['head-trainer'],
+    read : ['trainee', 'trainer'],
+    write : ['trainer'],
+    delete: [],
+    }
+    }
+    function HasPermission(ModuleName,role,PermissionType)
```
Comment (VisaaliSuccessive) [inline extra/utils/permissions.js]: please have a look into the camel casing :
"HasPermission" - "hasPermission"

File: extra/utils/validation.js (hunk 4 above)
Comment (VisaaliSuccessive) [reply extra/utils/validation.js]: need proper indentation

Comment (VisaaliSuccessive) [reply extra/utils/validation.js]: [Indentation Guidelines](https://github.com/airbnb/javascript )- use this for reference

File: extra/utils/permissions.js (hunk 8 above)
Comment (ajayrose25) [reply extra/utils/permissions.js]: Done

**Pull request merged**